File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        
        ArrayDeque<int[]> work = new ArrayDeque<>();
        ArrayDeque<Integer> finish = new ArrayDeque<>();
        
        for(int i=0; i<progresses.length; i++){
            work.add(new int[] {progresses[i], speeds[i]});
        }
        
        int index = 1;
        
        while(!work.isEmpty()){
            int cnt = 0;
            int size = work.size();
            for(int i=0; i<size; i++){
                int now[] = work.peek();
                if(now[0]+now[1]*index >= 100){
                    cnt++;
                    work.poll();
                }else break;
            }
            if(cnt > 0){
                finish.add(cnt);
            }
            index++;
        }
        
        int len = finish.size();
        int answer[] = new int[len];
        
        for(int i=0; i<len; i++){
            answer[i] = finish.poll();
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    
    const work = [];
    const finish = [];
    
    for(let i=0; i<progresses.length; i++){
        work.push([progresses[i], speeds[i]]);
    }
    
    var index = 1;
    
    while(work.length){
        var cnt = 0;
        var size = work.length;
        for(let i=0; i<size; i++){
            if(work[0][0] + work[0][1]*index >= 100){
                work.shift();
                cnt++;
            }else break;
        }
        if(cnt > 0){
            finish.push(cnt);
        }
        index++;
    }
    
    return finish;
}